earth_enterprise/src/common/serverdb/serverdbReader.cpp

```diff
@@ -29,6 +29,55 @@
 #include "common/geGdalUtils.h"
 #include "common/generic_utils.h"
 
+
+std::string JsToJson(const std::string& in)
+{
+  std::ostringstream oss;
+
+  const std::string alphaNumSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ_0123456789-abcdefghijklmnopqrstuvwxyz.";
+  const std::string valueSet = "0123456789.eE";  // .eE should only appear once but let something else enforce that
+
+  for (size_t i = 0; i < in.length(); ++i) {
+    if (isalpha(in[i])) {
+      static const std::array<std::string, 3> jsonLiterals = {"true", "false", "null"};
+      bool isLiteral = false;
+      for (const std::string& jlit : jsonLiterals) {
+        if (!jlit.compare(in.substr(i, jlit.size()))) {
+          oss << jlit;
+          i += jlit.size()-1;
+          isLiteral = true;
+        }
+      }
+
+      if (!isLiteral) {
+        size_t endEscape = in.find_first_not_of(alphaNumSet, i);
+        oss << "\"" << in.substr(i, endEscape-i) << "\"";
+        i = endEscape - 1;
+      }
+    }
+    else if (isdigit(in[i])) {
+      size_t endValue = in.find_first_not_of(valueSet, i);
+      oss << in.substr(i, endValue-i);
+      i = endValue - 1;
+    }
+    else if (in[i] == '\n') {
```

Comment:
I think we should remove this.  The output should preserve new lines of the original java script.